pkg/build/rpm_test.go

```diff
@@ -12,113 +13,252 @@ import (
-	rpmDestDir := builder.combustionDir
+	// Test
+	RPMFileNames, err := getRPMFileNames(RPMSourceDir)
+
+	// Verify
+	require.ErrorContains(t, err, "no RPMs found")
+
+	assert.Empty(t, RPMFileNames)
+}
+
+func TestCopyRPMsNoRPMDir(t *testing.T) {
+	// Setup
+	tmpDir, err := os.MkdirTemp("", "eib-copy-RPMs-test-no-dir-")
+	require.NoError(t, err)
+	defer os.RemoveAll(tmpDir)
 
-	file1Path := filepath.Join(rpmSourceDir, "rpm1.rpm")
-	defer os.Remove(file1Path)
+	bc := config.BuildConfig{}
+	builder := New(nil, &bc)
+	err = builder.prepareBuildDir()
+	require.NoError(t, err)
+
+	RPMSourceDir, err := builder.generateRPMPath()
+	require.NoError(t, err)
+
+	// Test
+	err = copyRPMs(RPMSourceDir, builder.combustionDir, nil)
+
+	// Verify
+	require.NoError(t, err)
+}
+
+func TestWriteRPMScript(t *testing.T) {
+	// Setup
+	tmpDir, err := os.MkdirTemp("", "eib-write-RPM-script-test-")
+	require.NoError(t, err)
+	defer os.RemoveAll(tmpDir)
+
+	RPMSourceDir := filepath.Join(tmpDir, "rpms")
+	err = os.Mkdir(RPMSourceDir, 0o755)
+	require.NoError(t, err)
+
+	buildConfig := config.BuildConfig{}
+	builder := New(nil, &buildConfig)
+	require.NoError(t, builder.prepareBuildDir())
+
+	file1Path := filepath.Join(RPMSourceDir, "rpm1.rpm")
 	file1, err := os.Create(file1Path)
 	require.NoError(t, err)
+	defer file1.Close()
 
-	file2Path := filepath.Join(rpmSourceDir, "rpm2.rpm")
-	defer os.Remove(file2Path)
+	file2Path := filepath.Join(RPMSourceDir, "rpm2.rpm")
 	file2, err := os.Create(file2Path)
 	require.NoError(t, err)
+	defer file2.Close()
+
+	RPMFileNames, err := getRPMFileNames(RPMSourceDir)
+	require.NoError(t, err)
 
 	// Test
-	err = builder.copyRPMs()
+	err = builder.writeRPMScript(RPMFileNames)
 
 	// Verify
 	require.NoError(t, err)
 
-	_, err = os.Stat(filepath.Join(rpmDestDir, "rpm1.rpm"))
+	expectedFilename := filepath.Join(builder.combustionDir, modifyRPMScriptName)
+	foundBytes, err := os.ReadFile(expectedFilename)
 	require.NoError(t, err)
 
-	_, err = os.Stat(filepath.Join(rpmDestDir, "rpm2.rpm"))
+	stats, err := os.Stat(expectedFilename)
 	require.NoError(t, err)
+	assert.Equal(t, fs.FileMode(modifyScriptMode), stats.Mode())
 
-	// Cleanup
-	assert.NoError(t, file1.Close())
-	assert.NoError(t, file2.Close())
+	foundContents := string(foundBytes)
+	assert.Contains(t, foundContents, "rpm1.rpm")
+	assert.Contains(t, foundContents, "rpm2.rpm")
 }
 
-func TestGetRPMFileNamesNoRPMs(t *testing.T) {
+func TestProcessRPMs(t *testing.T) {
 	// Setup
+	tmpDir, err := os.MkdirTemp("", "eib-process-RPMs-test-")
+	require.NoError(t, err)
+	defer os.RemoveAll(tmpDir)
+
+	RPMSourceDir := filepath.Join(tmpDir, "rpms")
+	err = os.Mkdir(RPMSourceDir, 0o755)
+	require.NoError(t, err)
+
 	bc := config.BuildConfig{
-		ImageConfigDir: "../config/testdata",
+		ImageConfigDir: tmpDir,
 	}
 	builder := New(nil, &bc)
-	err := builder.prepareBuildDir()
+	err = builder.prepareBuildDir()
 	require.NoError(t, err)
```

Comment:
Nit: All of this can be moved below to where it is actually used. Here we do it in the middle between creating the source dir and populating it with files.